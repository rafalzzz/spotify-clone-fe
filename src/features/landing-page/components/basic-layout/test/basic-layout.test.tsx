import { render, waitFor } from '@testing-library/react';

import { BasicLayout } from '..';

const mockedChildren = <div>Test</div>;

const renderLoveAlbumsSection = () => render(<BasicLayout>{mockedChildren}</BasicLayout>);

describe('BasicLayout', () => {
  it('render component without error', async () => {
    const { container } = renderLoveAlbumsSection();

    await waitFor(() => {
      expect(container).toBeDefined();
    });

    expect(container).toMatchSnapshot();
  });
});
