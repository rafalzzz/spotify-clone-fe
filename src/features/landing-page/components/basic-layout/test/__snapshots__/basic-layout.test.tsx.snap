// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayout render component without error 1`] = `
<div>
  <div
    class="basic-layout"
  >
    <div
      class="sidebar"
      data-testid="sidebar"
      style="width: 200px;"
    >
      <nav
        class="sidebar-navigation"
      >
        <ul>
          <li>
            <a
              class="sidebar-navigation-item"
              data-testid="sidebar-navigation-item"
              href="/"
            >
              <span
                aria-label="home"
                class="anticon anticon-home"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="home"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M946.5 505L534.6 93.4a31.93 31.93 0 00-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
                  />
                </svg>
              </span>
              Home
            </a>
          </li>
          <li>
            <a
              class="sidebar-navigation-item"
              data-testid="sidebar-navigation-item"
              href="/search"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
              Search
            </a>
          </li>
          <li>
            <a
              class="sidebar-navigation-item"
              data-testid="sidebar-navigation-item"
              href="/favorites"
            >
              <span
                aria-label="heart"
                class="anticon anticon-heart"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="heart"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                  />
                </svg>
              </span>
              Favorites
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="sidebar__resizer"
        data-testid="resizer"
      />
    </div>
    <div
      class="basic-layout__content"
    >
      <nav
        class="navbar"
      >
        <ul
          class="nav-buttons"
        >
          <li>
            <button
              class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-circle ant-btn-default nav-buttons__item"
              data-testid="undo-history"
              role="button"
              type="button"
            >
              <span
                aria-label="left"
                class="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li>
            <button
              class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-circle ant-btn-default nav-buttons__item"
              data-testid="redo-history"
              role="button"
              type="button"
            >
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </li>
        </ul>
        <ul
          class="redirects"
        >
          <li>
            <button
              class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-round ant-btn-default ant-btn-lg redirects__item sign-up-button"
              data-testid="redirection-to-registration"
              role="button"
              type="button"
            >
              <span>
                Sign up
              </span>
            </button>
          </li>
          <li>
            <button
              class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-round ant-btn-default ant-btn-lg redirects__item sign-in-button"
              data-testid="redirection-to-login"
              role="button"
              type="button"
            >
              <span>
                Sign in
              </span>
            </button>
          </li>
        </ul>
      </nav>
      <div>
        Test
      </div>
    </div>
    <footer
      class="footer-container"
    >
      <div
        class="now-played-track"
        data-testid="now-played-track"
      />
      <div
        class="player-controls"
      >
        <div
          class="player-buttons"
        >
          <button
            class="custom-icon-button"
            data-testid="shuffle-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="m4.5 6.8.7-.8C4.1 4.7 2.5 4 .9 4v1c1.3 0 2.6.6 3.5 1.6l.1.2zm7.5 4.7c-1.2 0-2.3-.5-3.2-1.3l-.6.8c1 1 2.4 1.5 3.8 1.5V14l3.5-2-3.5-2v1.5zm0-6V7l3.5-2L12 3v1.5c-1.6 0-3.2.7-4.2 2l-3.4 3.9c-.9 1-2.2 1.6-3.5 1.6v1c1.6 0 3.2-.7 4.2-2l3.4-3.9c.9-1 2.2-1.6 3.5-1.6z"
              />
            </svg>
          </button>
          <button
            class="custom-icon-button"
            data-testid="prev-song-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="M13 2.5L5 7.119V3H3v10h2V8.881l8 4.619z"
              />
            </svg>
          </button>
          <button
            class="custom-icon-button"
            data-testid="play-button"
            role="button"
          >
            <span
              aria-label="play-circle"
              class="anticon anticon-play-circle"
              data-testid="player-buttons-play-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="play-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
                />
              </svg>
            </span>
          </button>
          <button
            class="custom-icon-button"
            data-testid="next-song-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="M11 3v4.119L3 2.5v11l8-4.619V13h2V3z"
              />
            </svg>
          </button>
          <button
            class="custom-icon-button"
            data-testid="loop-button"
            role="button"
          >
            <svg
              class="svg-icon"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="svg-icon-path"
                d="M5.5 5H10v1.5l3.5-2-3.5-2V4H5.5C3 4 1 6 1 8.5c0 .6.1 1.2.4 1.8l.9-.5C2.1 9.4 2 9 2 8.5 2 6.6 3.6 5 5.5 5zm9.1 1.7-.9.5c.2.4.3.8.3 1.3 0 1.9-1.6 3.5-3.5 3.5H6v-1.5l-3.5 2 3.5 2V13h4.5C13 13 15 11 15 8.5c0-.6-.1-1.2-.4-1.8z"
              />
            </svg>
          </button>
        </div>
        <div
          class="music-progress-bar"
        >
          <audio />
          <button
            class="duration duration--text-align-end"
            data-testid="duration-button"
            disabled=""
          >
            0:00
          </button>
          <div
            class="progress-bar"
          >
            <input
              class="progress-bar__input"
              data-testid="progress-bar"
              max="0"
              min="0"
              step="0.01"
              type="range"
              value="0"
            />
            <span
              class="progress-bar__span-thumb"
              style="left: -3px;"
            />
          </div>
          <button
            class="duration duration--text-align-start"
            data-testid="duration-button"
          >
            0:00
          </button>
        </div>
      </div>
      <div
        class="sound-progress-bar"
      >
        <button
          class="custom-icon-button custom-icon-button--active"
          data-testid="sound-progress-bar-button"
          role="button"
        >
          <svg
            class="svg-icon"
            data-testid="sound-icon"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="svg-icon-path"
              d="M12.945 1.379l-.652.763c1.577 1.462 2.57 3.544 2.57 5.858s-.994 4.396-2.57 5.858l.651.763a8.966 8.966 0 00.001-13.242zm-2.272 2.66l-.651.763a4.484 4.484 0 01-.001 6.397l.651.763c1.04-1 1.691-2.404 1.691-3.961s-.65-2.962-1.69-3.962zM0 5v6h2.804L8 14V2L2.804 5H0zm7-1.268v8.536L3.072 10H1V6h2.072L7 3.732z"
            />
          </svg>
        </button>
        <div
          class="progress-bar"
        >
          <input
            class="progress-bar__input"
            data-testid="progress-bar"
            max="1"
            min="0"
            step="0.01"
            type="range"
            value="0.5"
          />
          <span
            class="progress-bar__span-thumb"
            style="left: -3px;"
          />
        </div>
      </div>
    </footer>
  </div>
</div>
`;
